test/models/financial_year_test.rb

```diff
@@ -149,20 +149,24 @@ class FinancialYearTest < ActiveSupport::TestCase
       result_journal: Journal.find_by(nature: :result, currency: f.currency) ||
                       Journal.create_one!(:result, f.currency)
     }
-    assert f.close(User.first, nil, options), "Financial year #{f.code} should be closed"
+    begin
+      assert f.close(User.first, nil, options), "Financial year #{f.code} should be closed"
 
-    assert f.prior_to_closure_archive.present?
-    assert f.post_closure_archive.present?
+      assert f.prior_to_closure_archive.present?
+      assert f.post_closure_archive.present?
 
-    assert File.exist? f.prior_to_closure_archive.path
-    assert File.exist? f.post_closure_archive.path
+      assert File.exist? f.prior_to_closure_archive.path
+      assert File.exist? f.post_closure_archive.path
 
-    assert File.exist? f.prior_to_closure_archive.path.gsub(/zip/, 'asc')
-    assert File.exist? f.post_closure_archive.path.gsub(/zip/, 'asc')
+      assert File.exist? f.prior_to_closure_archive.path.gsub(/zip/, 'asc')
+      assert File.exist? f.post_closure_archive.path.gsub(/zip/, 'asc')
 
-    crypto = GPGME::Crypto.new
-    assert_equal crypto.verify(f.prior_to_closure_archive.signature) { |s| s.valid? }.class, GPGME::Data
-    assert_equal crypto.verify(f.post_closure_archive.signature) { |s| s.valid? }.class, GPGME::Data
+      crypto = GPGME::Crypto.new
+      assert_equal crypto.verify(f.prior_to_closure_archive.signature) { |s| s.valid? }.class, GPGME::Data
+      assert_equal crypto.verify(f.post_closure_archive.signature) { |s| s.valid? }.class, GPGME::Data
+    ensure
```

Comment:
Cannot put in teardown as the financial year (`f`) is local to the test.
@Aquaj Is this OK? Should i look into refactoring this?

Reply:
It's okay. If we can't do it in `teardown` we'll keep each test that do it responsible for it for now. If this turns out to be repeated on several tests.